src/Events/Event.php

```diff
@@ -0,0 +1,208 @@
+
+        if ($random == 8) {
+            $this->merchantShipEvent($questlog, $hyperdrive);
+        }
+        if ($random == 9) {
+            $this->asteroidBeltEvent($player, $playerShip);
+        }
+        if ($random == 10) {
+            $this->spaceCombatEvent($player, $playerShip, $hyperdrive, $questlog, $surface, $event);
+        }
+
+    }
+
+
+    private function merchantShipEvent(QuestLog $questlog, HyperdriveNavigator $hyperdrive): void
+    {
+        $this->output->write("");
+        $this->output->write("You encountered a merchant's ship in need of repairs. They ask of you to deliver their cargo and you will be rewarded with half of their payment.");
+        $options = ["Yes" => "Yes", "No" => "No"];
+        $result = $this->output->getCli()->radio("Do you agree to help them and deliver their cargo?", $options)->prompt();
+        $this->output->write("");
+        if ($result === "Yes") {
+            $this->output->write("");
+            $this->output->write("You agreed to deliver their cargo.");
+            $this->output->write("");
+            $questlog->addQuest(new Quest(cargo: $questlog->getRandomCargo(), destination: $hyperdrive->getRandomPlanet(), completed: false, main: false, exp: 2500, reward: $questlog->generateReward()));
+        }
+        if ($result === "No") {
+            $this->output->write("");
+            $this->output->write("The merchant wasn't happy with your reply but you decided it's not your problem that he has a broken ship. You decide to leave him alone.");
+            $this->output->write("");
+        }
+    }
+
+    private function asteroidBeltEvent(Pilot $player, Ship $playerShip): void
+    {
+        $this->output->write("");
+        $this->output->info("You encountered an Asteroid belt! You lost some fuel while trying to maneuver through them.");
+        $fuelLost = 5 * (10 - $player->getSkill());
+        $playerShip->loseFuel($fuelLost);
+    }
+
+    private function spaceCombatEvent(Pilot $player, Ship $playerShip, HyperdriveNavigator $hyperdrive, QuestLog $questlog, PlanetSurface $surface, Event $event): void
+    {
+        $this->output->write("");
+        $this->output->info("You've been attacked!");
+        $enemies = new Enemies($this->output);
+        $combat = new Combat($this->output);
+        $combat->landingOrFighting($player, $playerShip, $enemies->getEnemyShips(), $surface, $event, $questlog, $hyperdrive);
+
+    }
+
+
+    public function randomLandEvents(Pilot $player, Ship $playerShip): void
+    {
+        $random = rand(1, 10);
+
+        if ($random == 7) {
+            $this->robberyEvent($player);
+        }
+        if ($random == 8) {
+            $this->reputationEvents($player, $playerShip);
+        }
+        if ($random == 9) {
+            $this->gamblingEvent($player);
+        }
+        if ($random == 10) {
+            $this->racingEvent($player);
+        }
+    }
+
+    private function robberyEvent(Pilot $player): void
```

Comment:
Tworzysz tutaj sporo metod, które wyglądają jakby mogły być spokojnie osobnymi klasami dziedziczącymi z klasy bazowej czy implementujący ten sam interfejs. `RobbertyEvent.php` `GamblingEvent.php`, itd.

Reply:
Czyli powinienem te metody wyrzucić do osobnych klas i zrobić z nich obiekty? Żeby w przypadku gdy gra wylosuje że ma się dziać event nr 8 to odpalało się np. new robberyEvent() ?

Reply:
Coś takiego. Fajnie by było, żeby były one pod jednym interfejsem czy tam klasą bazową.